src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1164,20 +1160,19 @@ internal ImmutableArray<TypeWithAnnotations> GetTypeParametersAsTypeArguments()
         /// symbol by type substitution then OriginalDefinition gets the original symbol as it was defined in
         /// source or metadata.
         /// </summary>
-        public new virtual NamedTypeSymbol OriginalDefinition
-        {
-            get
-            {
-                return this;
-            }
-        }
+        public new NamedTypeSymbol OriginalDefinition => _originalDefinition ?? GetAndSetOriginalDefinition();
```

Comment:
is this pattern substantively better than `_originalDefintion ??= OriginalDefinitionImpl`?

Reply:
Probably not; clearly I need to modernise my go to C# :)